models/brac/views/ancview_pregnancy.sql

```diff
@@ -0,0 +1,45 @@
+{{
+    config(
+        materialized = 'incremental',
+        indexes=[
+            {'columns': ['uuid']},
+            {'columns': ['danger_sign_at_reg']},
+            {'columns': ['reported']},
+            {'columns': ['"@timestamp"']}
+
+        ]
+    )
+}}
+
+    WITH config_cte AS (
+        SELECT ( value #>> '{lmp_calcs,maximum_days_pregnant}'::text[])::integer AS maximum_days_pregnant
+        FROM {{ ref("configuration") }}
+        WHERE key = 'anc'::text AND value ? 'lmp_calcs'::text
+        )
+    SELECT 
+        preg."@timestamp"::timestamp without time zone AS "@timestamp",
+        preg.uuid AS uuid,
+        'pregnancy'::text AS form,
+        preg.lmp <> '' AS has_lmp,
+        preg.lmp::date,
+        date_trunc('day', preg.reported::timestamp with time zone) <= (preg.lmp::date + '84 days'::interval) AS early_reg,
+        (preg.lmp::date + '84 days'::interval)::date AS first_tri_end,
+        (preg.lmp::date + '168 days'::interval)::date AS second_tri_end,
+        preg.edd, 
+        (preg.lmp::date + ((config.maximum_days_pregnant || ' days')::interval))::date AS mdd,
+        date_part('days', now() - preg.lmp::date::timestamp with time zone)::integer AS days_since_lmp,
+        date_part('days', preg.reported - preg.lmp::date::timestamp without time zone)::integer AS days_pregnant_at_reg,
+        preg.danger_signs <> ''::text AS danger_sign_at_reg,
+        preg.risk_factors <> ''::text AS has_risk_factor,
+        preg.imported,
+        preg.patient_id,
+        preg.chw AS reported_by,
+        contact.parent_uuid AS reported_by_parent,
+        preg.reported
+    FROM config_cte config,
+        {{ ref("useview_pregnancy") }} preg
+    INNER JOIN {{ ref("contactview_metadata") }} contact ON contact.uuid = preg.chw
+
+    {% if is_incremental() %}
+        AND preg . "@timestamp" > {{ max_existing_timestamp('"@timestamp"', target_ref=ref("useview_pregnancy")) }}
```

Comment:
Curious what your thought is on this situation:

New pregnancies are fetched since we pull based on the timestamp on useview_prepgnancy. Is there a chance that we could have more recent pregnancy information that contactview_metadata data.

I haven't thought much through this.... This incremental logic does make my head spin, sometimes

Reply:
```Is there a chance that we could have more recent pregnancy information that contactview_metadata data.``` This seems incomplete. I am not sure what the question is here.

Could you explain this question in a simpler way.

Reply:
Updated _that_ to _than_.

My initial comment might be a NOOP at this time.